build-scripts/xc-universal-binary.sh

```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env bash
 set -euvx
 
+# XCode tries to be helpful and overwrites the PATH. Reset that.
+PATH="$(bash -l -c 'echo $PATH')"
```

Comment:
Have we tested that this doesn't break anything locally?

Reply:
WFM locally.

Reply:
This worries me that we're going to be using a different clang than the one bundled with xcode that swift links against (swift links against libclang for various reasons like parsing C++ headers and such), but maybe it's fine...

Reply:
Agree, I'm not a huge fan of the solution either but I couldn't find anything wrong with it so far.